modules/sdk/src/main/scala/org/tessellation/sdk/infrastructure/cluster/services/Session.scala

```diff
@@ -29,6 +34,20 @@ object Session {
             sessionStorage.createToken
           }
 
-      def verifyToken(peer: PeerId, headerToken: Option[SessionToken]): F[TokenVerificationResult] = ???
+      def verifyToken(peer: PeerId, headerToken: Option[SessionToken]): F[TokenVerificationResult] =
+        headerToken.fold[F[TokenVerificationResult]](EmptyPeerToken.pure[F]) { token =>
```

Comment:
I guess that's rather `EmptyHeaderToken`. If I remember correctly `EmptyPeerToken` was meant for the case where we have a peer connected but we got None back when fetching that peer's token from `TokenService`. Now this case may actually not be possible anymore, so I would clean up the `TokenVerificationResult` reasons.

Reply:
:+1: